### PR TITLE
Stats: Persist highlights setting notice status

### PR DESCRIPTION
**Pull request by kangzj**
Related to #77548 

## Proposed Changes

The PR proposes to query notices status with `react-query` and calls API to persist the dismissed status for Highlights settings tooltip.

## Testing Instructions
* Apply D112871-code to your sandbox
* Sandbox `public-api.wordpress.com` 
* Open the Traffic page on the Calypso Live site
* Apply feature flag `?flags=stats/highlights-settings`
* Ensure you see the tooltip for Highlights settings

* Click 'Got it' or the hamburger icon
* Ensure the tooltip is dismissed
* Refresh the page
* Ensure the tooltip doesn't come up again

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/hooks/use-notice-visibility-query.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+export function queryNotices( siteId: number | null ): Promise< any > {
```
Comment (dognose24) [inline client/my-sites/stats/hooks/use-notice-visibility-query.ts]:
```suggestion
export function queryNotices( siteId: number | null ): Promise< Notices > {
```

```
type Notices = {
	new_stats_feedback: boolean;
	opt_in_new_stats: boolean;
	opt_out_new_stats: boolean;
	traffic_page_highlights_module_settings: boolean;
	traffic_page_settings: boolean;
};
```

What do you think about introducing the type `Notices` to demonstrate keys from the API response as well?

**Review by dognose24 — APPROVED** [review]
Works as described! @kangzj Nice Work! 👍🏼 I just left the nit comment that won't be a blocker. 🙂

**Review by dognose24 — APPROVED** [review]
Works as described! @kangzj Nice Work! 👍🏼 I just left the nit comment that won't be a blocker. 🙂

File: client/my-sites/stats/hooks/use-notice-visibility-query.ts (hunk 1 above)
Comment (kangzj) [reply client/my-sites/stats/hooks/use-notice-visibility-query.ts]: Addressed in https://github.com/Automattic/wp-calypso/pull/77867/commits/c13e2ed44ff6d048dd3be9e52b34df74d7bf5c85

**Pull request merged**